### ISSUE TITLE
streamline sessionname

### ISSUE THREAD

**Issue by yakir12** [jpalardy/vim-slime#214]
This is not an issue, more of a help request, but should prove useful for others as well.

My work process is:
1. navigate to the directory where my project is (I use `ranger`)
2. spawn a terminal (`xterm`)
3. start a (in my case) Julia session (but could be any programming language) with:
```
screen -S <X> julia
```
where `<X>` is a letter I know isn't used by an already existing screen session (I move up the alphabet from `a` to `b`, `c`, etc.)
4. open the main script-file in that directory (I use `neovim`)
5. `C-c, C-c` and manually specify the same `sessionname` as I did before (I always use "0" for the `windowname`).

It would be great if there was a way to automate this:
1. spawn a terminal with a screen session with a novel `sessionname`
2. open a file with a (neo)vim where the `sessionname` in `vim-slime` is already specified to that novel `sessionname` from the previous step (and a default `windowname` of 0). 

This would streamline the whole process immensely.

I realize this is not `vim-slime` specific, ultimately it involves `screen`, `bash`, and maybe `ranger`, but I thought it's well worth asking...

Thanks for a great tool!

**Comment by jpalardy** [comment]
Hi there,

Yes, I've struggled also with having to specify the target window. It often feels like ceremony when it's "obvious" from context. (for humans…)

If I could convince you to try tmux instead:

> let g:slime_default_config = {"socket_name": "default", "target_pane": "{right-of}"}
([link](https://github.com/jpalardy/dotfiles/blob/master/vim/plugins.vim#L24))

The default socket, and "right-of" mean, in practice, that I double-tap <kbd>enter</kbd> when asked for config.

If you decide to stick to GNU screen, you could set an environment variable with a script (whatever logic is needed) and use that in your vim config to set `g:slime_default_config` to what you need.

Let me know how that goes.

**Comment by yakir12** [comment]
Oh, I'll try tmux instead of screen, no problem. Just thought that since screen seems to be the default for this plugin, screen would be the way to go. 

I'll report back. Thanks for helping!

**Comment by yakir12** [comment]
OK, my "problem" with `tmux` is that I live in `i3wm` and I like to manage all my windows (read panes) with `i3`. I therefore don't like that `tmux` manages my panes. So things like `"{right-of}"` don't work for me (cause the terminal I want to send things to is in another window).

Hmmm...

**Comment by jpalardy** [comment]
oh, I see… interesting

Before {right-of}, I ran:

```bash
> echo $TMUX_PANE
%52
```

inside my .bashrc, so that every shell that ran tmux (any possible target) would "show me its number" on startup (to be used, in this case, instead of {right-of})

Alternatively, whether screen or tmux, I imagine it could be possible to write a script to figure out:

- what's the latest pane I opened?
- what pane is running a (relevant) REPL?
- or something else…

I personally haven't found a recipe that would _always_ get me what I want. (the REPL I'm LOOKING at! 😄)
